posix/unix.c

```diff
@@ -567,6 +582,12 @@ int unix_connect(unsigned socket, const struct sockaddr *address, socklen_t addr
 			hal_spinlockSet(&s->spinlock, &sc);
 			s->state |= US_CONNECTING;
 
+			if (s->nonblock && (s->connect == NULL)) {
+				hal_spinlockClear(&s->spinlock, &sc);
+				err = -EINPROGRESS;
```

Comment:
Please note that in this case we should handle `SO_ERROR` in `unix_getsockopt()` to allow user to detrmine whether `connect()` succeeded or failed (see man page on `connec()t` for details).

Reply:
Done